velox/substrait/SubstraitToVeloxPlan.h

```diff
@@ -127,6 +160,27 @@ class SubstraitVeloxPlanConverter {
       const ::substrait::Expression& substraitFilter,
       std::vector<::substrait::Expression_ScalarFunction>& scalarFunctions);
 
+  /// Helper function to extract the attributes required to create a ProjectNode
+  /// used for interpretting Substrait Emit.
+  template <typename RelMessage>
+  EmitInfo GetEmitInfo(const RelMessage& rel, const core::PlanNodePtr& node) {
```

Comment:
naming: GetEmitInfo -> getEmitInfo

Does this need to be a template?

Reply:
We could pass any Substrait relation here, and they don't have any base-class. I felt like a template is required. Any better suggestion?

Reply:
Assuming 'emit' variable has a specific type, we can modify this method to take 'emit' instead of 'rel' and modify the caller to pass rel.common().emit() instead of rel. Then, this method won't need to be a template.

Reply:
That's fair. 👍

Reply:
I removed the templated functions.

Reply:
Now `RelCommon` is used instead of a templated value.